public/js/queue.js

```diff
@@ -242,22 +242,26 @@
     };
 
     // recent songs
-    QP.getRecent = function(){
-      var This = this;
-
-
-      $.get("/recent", function(data){
-        if(data.length > 0){
-          $('#recent').remove();
-          $('.data').append('<div id="recent" class="span4"><h2>Recent</h2><ol></ol></div>');
-
-          html_recent = '';
-
-        //
-        This.loadRecent(0,(data.length - 1),data);
+    QP.getRecent = async() => {
+      try {
+        await(this.getQueue);
+        await(this.getNowPlaying);
```

Comment:
This would fetch data from the end points one by one, which will slower the page loading _(won't make much difference but still)_. A better way to solve the problem would be to use [`Promise.all()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). This will give you all fetched data in an array (in order of the promises passed, regardless of completion order) which you can then utilize to render the page.